### PR TITLE
fliplet test

### PR DESCRIPTION
**Pull request by seromenho**
- Add fliplet tests runner
- For now it works for widgets
- Included test suite on the widget template boilerplate

### CODE REVIEW

File: docs/Testing-components.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+## Where should my tests go
+
+You should include your tests under `tests` folder of the widget.
+
+## Available to write your tests
+
+- interfaceBrowser - A Nightmare instance of your widget interface.
+- buildBroser - A Nightmare instance of your widget instance build.
```
Comment (tonytlwu) [inline docs/Testing-components.md]: Typo: buildBrowser

File: fliplet-test.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+let app;
+let page;
+let widgetInstance;
+
+// Change package name to avoid using an existing one on the environment
+const fileName = './widget.json';
+const file = require(`${process.cwd()}/widget.json`);
+const packageName = file.package;
+const tags = file.tags;
+file.package = `${casual.domain}.${casual.unix_time}`;
```
Comment (squallstar) [inline fliplet-test.js]: this is better to be `test.${packageName}.${casual.unix_time}`

File: fliplet-test.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+          return file.substr(-3) === '.js';
+        }).forEach(function(file){
+          mocha.addFile(
+            path.join(testDir, file)
+          );
+        });
+
+        mocha.run()
+          .on('end', function () {
+            restoreWidgetJson();
```
Comment (squallstar) [inline fliplet-test.js]: I think we should restore the json as soon as the widget is published. Is it possible to do that?

File: docs/Testing-components.md (hunk 1 above)
Comment (seromenho) [inline docs/Testing-components.md]: Fixed eagle eye.

File: fliplet-test.js (hunk 2 above)
Comment (seromenho) [inline fliplet-test.js]: It is. Changed

File: fliplet-test.js (hunk 3 above)
Comment (seromenho) [inline fliplet-test.js]: Yes, definitely. Done.

**Comment by seromenho** [comment]
@tonytlwu @squallstar requested changes done.

**Pull request merged**